### PR TITLE
feat: added drawer component

### PR DESCRIPTION
**Pull request by GuptaEesh**
# types are added:

>Simple drawer
Permanent drawer
Dismissable Drawer

IMG_20220212_194707

### CODE REVIEW

File: components/drawer.css [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    flex: 1;
+    flex-direction: column;
+    gap: var(--size-14);
+    height: 100%;
+    max-width: 12rem;
+    padding: var(--size-12);
+    overflow-y: auto;
+}
+
+.title-for-navigation {
```
Comment (saadahmadhere) [inline components/drawer.css:17]: these kind of names could be shorter. e.g. title-navigation

**Review by saadahmadhere — COMMENTED** [review]
Code looks cleans and good use of comments.

File: components/drawer.css (hunk 1 above)
Comment (GuptaEesh) [reply components/drawer.css:17]: Yeah right! I will look into it after merging. Still a little hazy about naming stuff.

Comment (MaruthiPothuganti) [inline components/drawer.css:17]: Have Small class names which can convey the meaning like .nav-title for this and .drawer-nav for above etc.

**Review by MaruthiPothuganti — APPROVED** [review]
Code looks clean. Suggested a small change

File: components/drawer.css (hunk 1 above)
Comment (GuptaEesh) [reply components/drawer.css:17]: Yeah right! I will look into it after merging. Still a little hazy about naming stuff. I will definitely work on this.

**Comment by GuptaEesh** [comment]
Insightful comments. Merging this branch now.

**Pull request merged**